cloud-gcp/src/main/java/com/sequenceiq/cloudbreak/cloud/gcp/GcpProvisionSetup.java

```diff
@@ -146,4 +139,111 @@ private boolean containsSpecificImage(ImageList imageList, String imageUrl) {
         }
         return false;
     }
+
+    private boolean bucketExist(Storage storage, String bucketName) {
+        boolean existingBucket;
+        try {
+            storage.buckets().get(bucketName).execute();
+            existingBucket = true;
+        } catch (GoogleJsonResponseException ex) {
+            existingBucket = false;
+            if (ex.getStatusCode() == NOT_FOUND) {
+                LOGGER.warn("Bucket {} does not exist on provider side so we will create it: {}",
+                        bucketName, ex.getMessage());
+            } else {
+                LOGGER.warn("We were not able to get the bucket from Google side with name {} with exception {}. "
```

Comment:
in this case are we going to create the bucket?

Reply:
yes, as the `existingBucket` boolean remains false.